src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -417,7 +417,9 @@ private SolutionState AddProject(ProjectId projectId, ProjectState projectState)
 
             var newProjectIds = _projectIds.ToImmutableArray().Add(projectId);
             var newStateMap = _projectIdToProjectStateMap.Add(projectId, projectState);
-            var newDependencyGraph = CreateDependencyGraph(newProjectIds, newStateMap);
+            var newDependencyGraph = _dependencyGraph
+                                        .WithAdditionalProjects(SpecializedCollections.SingletonEnumerable(projectId))
+                                        .WithAdditionalProjectReferences(projectId, projectState.ProjectReferences.Select(r => r.ProjectId).ToList());
```

Comment:
instead of .ToList.  could you just use SelectAsArray and pass in an ImmutableArray?